src/reducers/listByIdReducer.ts

```diff
@@ -0,0 +1,32 @@
+import { requestStatus } from '../types/index';
+import { FETCH_LIST_SUCCESS, FETCH_LIST_START, FETCH_LIST_ERROR } from '../actions/fetchList';
+export type ListActions = 'FETCH_LIST_SUCCESS' | 'FETCH_LIST_START' | 'FETCH_LIST_ERROR';
+export interface ListState {
+  listData: any;
+  getListStatus: requestStatus;
+}
+// eslint-disable-next-line @typescript-eslint/class-name-casing
+interface listAction {
+  type: ListActions;
+  payload?: any;
+}
+
+const initialState = {
+  listData: [],
+  getListStatus: requestStatus.loading,
+};
+
+const listByIdReducer = (state = initialState, action: listAction): ListState => {
+  switch (action.type) {
+    case FETCH_LIST_START:
+      return { ...state, getListStatus: requestStatus.loading };
+    case FETCH_LIST_SUCCESS:
+      return { listData: action.payload, getListStatus: requestStatus.success };
+    case FETCH_LIST_ERROR:
+      return { ...state, getListStatus: requestStatus.error };
+    default:
+      return state;
+  }
+};
+
+export default listByIdReducer;
```

Comment:
This reducer should follow the pattern we're using in the `trips` reducer.

```
initialState = {
  byId: {},
  getListStatus: requestStatus.loading,
}
```

Then we should listen for the `CREATE_TRIP_SUCCESS` action type.

We know the payload for that action has a list object in it.

```
case CREATE_TRIP_SUCCESS
  return { byId: { ...state.byId, [action.payload.listId]: action.payload.list } };
```

This way, when rendering the trip/list page, we don't need to fetch the list if it's already there 

```
  useEffect(() => {
    if (!listsById[currentTrip.listId]) {
      fetchList(currentTrip.listId);
    }
  }, [id]);


...
// inside map state to props
  listsById: state.lists.byId, // we also need to change the redux store key to the plural 'lists'.
```